Problem3/src/StringCompressor.java

```diff
@@ -0,0 +1,29 @@
+import java.util.ArrayList;
+
+//TASK:
+//Implement a method to perform basic string compression using the counts
+//of repeated characters. For example, the string "aabcccccaaa" would become
+//"a2b1c5a3". If the compressed string would not become smaller than
+//the original string, your method should return the original string.
+
+public class StringCompressor {
+	
+	public String compressString (String inputString){
+		
+		char [] charsFromString = inputString.toCharArray();
+		StringBuilder sb = new StringBuilder ();
+
+		int counter = 1;
+		for (int i = 0; i < charsFromString.length; i++){
+			if ((i!=charsFromString.length-1) && (charsFromString[i]==charsFromString[i+1])){
+				counter++;
+			}
+			else {
+				sb.append(String.valueOf(charsFromString[i]));
+				sb.append(String.valueOf(counter));
+				counter=1;
+			}
+		}
+		return (inputString.length()>sb.length())?sb.toString():inputString;
+	}
+}
```

Comment:
Good implementation. Two general comments. Pelase provide unit tests.
Teh second comment. Let's re-implement this task with java streams as one more implementation (the same idea - via java interface and several impementations)